flash/image/segmentation/heads.py

```diff
@@ -33,18 +35,20 @@
     def _load_smp_head(
         head: str,
         backbone: str,
-        pretrained: bool = True,
+        pretrained: Union[bool, str] = True,
         num_classes: int = 1,
         in_channels: int = 3,
         **kwargs,
-    ) -> Callable:
+    ) -> nn.Module:
 
         if head not in SMP_MODELS:
             raise NotImplementedError(f"{head} is not implemented! Supported heads -> {SMP_MODELS.keys()}")
 
         encoder_weights = None
-        if pretrained:
+        if pretrained is True:
             encoder_weights = "imagenet"
+        elif isinstance(pretrained, str):
+            encoder_weights = pretrained
```

Comment:
Maybe better to switch:
```suggestion
        if isinstance(pretrained, str):
            encoder_weights = pretrained
        elif pretrained:
            encoder_weights = "imagenet"
```